browser/src/control/Control.JSDialog.js

```diff
@@ -383,15 +383,24 @@ L.Control.JSDialog = L.Control.extend({
 		if (instance.canHaveFocus && initialFocusElement && initialFocusElement.length)
 			initialFocusElement[0].focus();
 
-		var focusWidget = instance.init_focus_id ? instance.container.querySelector('[id=\'' + instance.init_focus_id + '\']') : null;
-		if (focusWidget)
-			focusWidget.focus();
-		if (focusWidget && document.activeElement !== focusWidget) {
-			var firstFocusable = JSDialog.GetFocusableElements(focusWidget);
-			if (firstFocusable && firstFocusable.length)
-				firstFocusable[0].focus();
-			else
-				console.error('cannot get focus for widget: "' + instance.init_focus_id + '"');
+		const instanceContent = instance.content; // instance.content holds the HTML of the current dialog
+		const tabControl = instanceContent.querySelector('#tabcontrol');
```

Comment:
This still uses ID :) please check the JSON, 'type' field. This is misleading a bit instance keeps at the same time JSON and DOM elements here after some rework.
This is important as not always tabcontrol will have such id. Please look for children and traverse the tree of widgets for that type of widget.